proposals/042-health-checks.md

```diff
@@ -0,0 +1,236 @@
+
+For Syndesis we would need to be in full control, and therefore we will need to ensure Apache Camel can startup even if there is a connection problem with a remote system. This requires to configure Apache Camel to not auto start the routes, which can be done in Spring Boot by configuring `application.properties` with the following:
+
+    camel.springboot.auto-startup = false
+
+By doing so no routes is automatic startup up, but we would then need to defer starting these routes and do this using another way.
+
+When using spring,the camel context can be configured using beans from the registry [1] therefore we can add a new auto discovered `RouteController` SPI which will be the responsible for starting up the routes in a controlled manner.
+
+NOTE: as the idea of a route controller has been brought up for Camel 3.0, the  RouteController will be an early preview SPI.
+
+NOTE: as refactoring the route mamagement requires to change camel-core, the first implementation leverages `RoutePolicy`/`RoutePolicyFactory`
+
+The `RouteController` implementation will sets `autoStartup = false` and will install its own `RoutePolicyFactory` which then creates a `RoutePolicy`  for each route an where we in the `onInit` callback can register the routes to the `RouteController` which then runs as a background thread and orchestrates starting the routes. The `RouteController` will then be responsbible for starting up all these routes, and have support for periodically retry in case a route fails to startup. And only when all routes has been started the `RouteController` has completed starting up all the routes.
+
+In Camel with Spring Boot we can setup this in the `Main` class via something along the following code lines:
+
+```java
+    @Bean
+    RouteController newRouteController(CamelContext camelContext) {
+        // The route controller is also in charge to add add its own
+        // RoutePolicyFactory to the camel context
+        return new RouteController(camelContext);
+    }
+```    
+
+
+TODO:
+- We can then implement a `DefaultRouteController` in camel-core which in 3.0 which will take over the current route management and then one can either configure its behavior or implement its own using DefaultRouteController as base class.
+- The syndesis `RouteController` should have configuration setting to specify how often to retry starting failed routes (we may need to have backoff etc).
+
+
+Besides starting up routes the `RouteController` should have a Java, JMX API, and REST which can report back status of the running integrations (i.e. running routes) and should also have APIs to force starting a route on demand, eg so an user can click a button, and then it will immediately schedule to run a task that attempts to start the route.
```

Comment:
As we bundle Jolokia (at least I think we do) is JMX API enough? We can reuse Jolokia authentication, so that would be a plus.